stl/inc/algorithm

```diff
@@ -2668,6 +2668,46 @@ namespace ranges {
     };
 
     inline constexpr _Find_first_of_fn find_first_of{_Not_quite_object::_Construct_tag{}};
+
+    // ALIAS TEMPLATE swap_ranges_result
+    template <class _In1, class _In2>
+    using swap_ranges_result = in_in_result<_In1, _In2>;
+
+    // VARIABLE ranges::swap_ranges
+    class _Swap_ranges_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2>
+            requires indirectly_swappable<_It1, _It2>
+        constexpr swap_ranges_result<_It1, _It2> operator()(
+            _It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2) const {
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+            auto _UFirst1 = _Get_unwrapped(_STD move(_First1));
+            auto _ULast1  = _Get_unwrapped(_STD move(_Last1));
+            auto _UFirst2 = _Get_unwrapped(_STD move(_First2));
+            auto _ULast2  = _Get_unwrapped(_STD move(_Last2));
+            for (; _UFirst1 != _ULast1 && _UFirst2 != _ULast2; (void) ++_UFirst1, (void) ++_UFirst2) {

```

Comment:
We put the `(void)` casts in comma expressions that involve user-defined types because the comma operator is overloadable in C++ but we want to avoid calling any user overloads for these types. A function can't have a parameter of `void` type, so there can be no matching overload that accepts the result of the `(void)` cast. Further, since `,` is a binary operator, we don't need to `(void)`-cast every operand in a comma expression: it suffices to cast every second operand because either operand having type `void` is enough. Consequently, we only `(void)` cast every second operand in a comma series:
```suggestion
            for (; _UFirst1 != _ULast1 && _UFirst2 != _ULast2; ++_UFirst1, (void) ++_UFirst2) {
```